src/search/randomdestinationairportpickingbycriteria.h

```diff
@@ -0,0 +1,41 @@
+#ifndef RANDOMDESTINATIONAIRPORTPICKINGBYCRITERIA_H
+#define RANDOMDESTINATIONAIRPORTPICKINGBYCRITERIA_H
+
+#include "search/randomdepartureairportpickingbycriteria.h"
+
+#include <QObject>
+#include <QThread>
+
+namespace atools {
+  namespace geo {
+    class Pos;
+  }
+}
+
+class RandomDestinationAirportPickingByCriteria : public QThread
+{
+  Q_OBJECT
+
+public:
+  RandomDestinationAirportPickingByCriteria(int indexDeparture);
+
+  // required calling !!
+  static void initStatics(const int &countResult, const int &randomLimit, std::pair<int, atools::geo::Pos>* data, const int &distanceMin, const int &distanceMax);
+
+  void run() override;
+
+signals:
+  void resultReady(const bool isSuccess, const int indexDeparture, const int indexDestination);
+
+private:
+  friend class RandomDepartureAirportPickingByCriteria;
+  static bool stopExecution;                                                    // Qt has no way to finally stop a thread
+  int indexDeparture;
```

Comment:
Only quickly looked over it and therefore do not understand all the code in detail but I hope that all variable access is gated by mutexes or similar constructs.